src/filters.rs

```diff
@@ -197,3 +200,38 @@ impl ResolveFilter for LowerFilter {
         Ok(content)
     }
 }
+
+#[derive(Debug)]
+pub struct SlugifyFilter;
+
+impl ResolveFilter for SlugifyFilter {
+    fn resolve<'t, 'py>(
+        &self,
+        variable: Option<Content<'t, 'py>>,
+        py: Python<'py>,
+        _template: TemplateString<'t>,
+        _context: &mut Context,
+    ) -> TemplateResult<'t, 'py> {
+        let non_word_re = Regex::new(r"[^\w\s-]").unwrap();
+        let whitespace_re = Regex::new(r"[-\s]+").unwrap();
+
+        let content = match variable {
+            Some(content) => {
+                let content = content.to_py(py).str()?.extract::<String>()?;
```

Comment:
This feels inefficient to me - we're converting to a Python object and then back to `String`. Can we do:

```suggestion
                let content = match content.resolve_string()?.content();
```

instead? Or is there a reason this is invalid?